homeassistant/components/xiaomi_aqara/config_flow.py

```diff
@@ -0,0 +1,167 @@
+"""Config flow to configure Xiaomi Aqara."""
+import logging
+
+import voluptuous as vol
+from xiaomi_gateway import XiaomiGatewayDiscovery
+
+from homeassistant import config_entries
+from homeassistant.const import CONF_HOST, CONF_MAC, CONF_NAME, CONF_PORT
+from homeassistant.helpers.device_registry import format_mac
+
+# pylint: disable=unused-import
+from .const import (
+    CONF_INTERFACE,
+    CONF_KEY,
+    CONF_PROTOCOL,
+    CONF_SID,
+    DOMAIN,
+    ZEROCONF_GATEWAY,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+DEFAULT_GATEWAY_NAME = "Xiaomi Aqara Gateway"
+DEFAULT_INTERFACE = "any"
+
+
+GATEWAY_CONFIG = vol.Schema(
+    {vol.Optional(CONF_INTERFACE, default=DEFAULT_INTERFACE): str}
+)
+GATEWAY_SETTINGS = vol.Schema(
+    {
+        vol.Optional(CONF_KEY): vol.All(str, vol.Length(min=16, max=16)),
+        vol.Optional(CONF_NAME, default=DEFAULT_GATEWAY_NAME): str,
+    }
+)
+
+
+class XiaomiAqaraFlowHandler(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a Xiaomi Aqara config flow."""
+
+    VERSION = 1
+    CONNECTION_CLASS = config_entries.CONN_CLASS_LOCAL_PUSH
+
+    def __init__(self):
+        """Initialize."""
+        self.host = None
+        self.interface = DEFAULT_INTERFACE
+        self.gateways = None
+        self.selected_gateway = None
+
+    async def async_step_user(self, user_input=None):
+        """Handle a flow initialized by the user."""
+        errors = {}
+        if user_input is not None:
+            self.interface = user_input[CONF_INTERFACE]
+
+            # Discover Xiaomi Aqara Gateways in the netwerk to get required SIDs.
+            xiaomi = XiaomiGatewayDiscovery(self.hass.add_job, [], self.interface)
+            await self.hass.async_add_executor_job(xiaomi.discover_gateways)
```

Comment:
If the interface address is badly formatted or incorrect somehow there will probably be an error here that we should catch.

Reply:
Nice, you are completly right!
I have added the exception, handeld the error and added a test for it.